p2p/peer.go

```diff
@@ -724,6 +719,16 @@ func (m *PeerManager) Dialed(peerID NodeID, address PeerAddress) error {
 
 	delete(m.dialing, peerID)
 
+	var upgradePeer NodeID
+	for from, to := range m.upgrading {
+		if to == peerID {
+			delete(m.upgrading, from)
+			upgradePeer = from
+			// Don't break, just in case this peer was marked as upgrading for
+			// multiple lower-scored peers (shouldn't really happen).
```

Comment:
How is this possible if we have a map for upgrading?

Reply:
Because we're comparing on the map value, not the map key. We want to delete all the entries where the value (the peer upgraded to) is the peer we've dialed.